### ISSUE TITLE
Is this still maintained ?

### ISSUE THREAD

**Issue by leonardo2204** [roughike/BottomBar#606]
I've seen a lot of open issues but no response from the author.
Some PR`s waiting to be reviewed also... 
So, is this project still alive ??

**Comment by searchy2** [comment]
This repo has the most number of unapproved PRs I have ever seen. A lot of the PRs would really improve this library's functionality. Too bad it appears no one is merging them!

**Comment by eriuzo** [comment]
perhaps he sees no point on maintaining it since we have https://developer.android.com/reference/android/support/design/widget/BottomNavigationView.html now

**Comment by leonardo2204** [comment]
@eriuzo Although I agree that Google released it's own version of the bottom bar, it is pretty bad right now, doesn't have like 10% of the functionalities this lib does.
So, deprecating this now would seem a little bogus, I just think he lets it go.

**Comment by leonardo2204** [comment]
Anyway, my point is, this lib is pretty cool, so IMHO, if the author does not have time to continue taking care of it, maybe adding more contributor to allow accepting PR's would be awesome !

**Comment by kassim** [comment]
Sephiroph74's fork seems to be relatively well maintained https://github.com/sephiroth74/Material-BottomNavigation - I'll see if it works for my needs

**Comment by searchy2** [comment]
I am using https://github.com/Ashok-Varma/BottomNavigation which is very similar to this library and slightly better maintained :dango:

**Comment by searchy2** [comment]
@leonardo2204 You would really think that Google with its billions could afford to create some decent UI libraries to make Android development easier. Apple provides a ridiculously intuitive drag and drop development interface and many prebuilt components that actually work. Why can't Google do the same?

**Comment by LuisHCK** [comment]
This seems nice
https://github.com/sephiroth74/Material-BottomNavigation
i forget this projects... Maybe the project is unmanteined

**Comment by leonardo2204** [comment]
I agree with all your comments, but my point is, this is a cool lib, with almost 6k stars, so it's obviously ok if the author don't have the time or don't want to maintain this anymore.  In that case, adding some people as contributors, therefore able to accept PR's and close issues, would be great to keep the library alive.

**Comment by leonardo2204** [comment]
From recent merges I guess @roughike is back!
Thanks

**Comment by roughike** [comment]
Yay!

I had a kind of burnout because of this very project. Time for a little novel:

When I made [the first version](https://github.com/roughike/BottomBar/tree/v1) I had no idea about automated testing and as a result, my code wasn't very clean. I regularly coded tens of hours in couple days and it became more and more spaghetti code over time. Because there were no automated tests in place, stuff was breaking all over. Bugs I had just fixed reappeared when fixing other bugs. Good times.

Then I had a break because I just was stressing too much. After revisiting the project after couple weeks, there was hundred or so new issues. When trying to fix them I realized I had no idea what the code did. So I started over and made the second version, trying to focus on testable and cleaner code. But in retrospect, I tried to do too much stuff at once and fix the whole project in one week or so. That resulted in a break that took three or so months.

But now I'm back. I'm trying to see past the horrible number of open issues and focus on small changes at a time. I'll check and fix stuff in a slow but steady pace, as doing too much at once just would result in another rage quit, as I learned.

TL;DR:
* was a noob to testing and made spaghetti code
* wrote spaghetti code on top of spaghetti code
* created so much spaghetti that everything broke when fixing bugs
* got burnt out and had a long break
* now back, wiser than ever

Eating a months worth of spaghetti at once tears your asshole. Smaller portions make your pooping experience far more enjoyable.

**Comment by roughike** [comment]
Reopening this one so others can see it too.

**Comment by leonardo2204** [comment]
@roughike Hahaha sorrry to laugh at it, but that was a funny yet sad history.
Glad you are back, hope to contribute even more with the library !

**Comment by LuisHCK** [comment]
You taught me a lesson, no more spaghetti code in my projects.

**Comment by pollax** [comment]
@searchy2, @eriuzo and others: 
I guess one way to help reducing the backlog is to answer/comment the issues that are open. Many issues are asking for very specific features which I doubt BottomBar does not intend to support, at least not in the foreseeable.

This would make it easier for @roughike and contributors to simply scan the issues and close a lot of "low hanging fruit".
